lua/entities/gmod_wire_expression2/core/egpobjects.lua

```diff
@@ -0,0 +1,493 @@
+--
+--	File for EGP Object handling in E2.
+--
+
+local function Update(self, this)
+	self.data.EGP.UpdatesNeeded[this] = true
+end
+
+---- Type defintion
+
+registerType("egpobject", "xeo", nil,
+	nil,
+	nil,
+	function(retval)
+		if retval == nil then return end
+		if not istable(retval) then error("Return value is neither nil nor a table, but a " .. type(retval) .. "!", 0) end
+	end,
+	function(v)
+		return not istable(v)
+	end
+)
+
+__e2setcost(2)
+
+registerOperator("ass", "xeo", "xeo", function(self, args)
+	local lhs, op2, scope = args[2], args[3], args[4]
+	local rhs = op2[1](self, op2)
+	if rhs == nil then return nil end
+	
+	local Scope = self.Scopes[scope]
```

Comment:
This middle lookup code is only necessary for types that can have their internals modified by E2 and will cause triggers on these changes (ie table, array, vector, angle)